tinygrad/shape/symbolic.py

```diff
@@ -77,7 +77,7 @@ def __mod__(self, b:int):
     assert b > 0
     if b == 1: return NumNode(0)
     if self.min >= 0 and self.max < b: return self
-    if self.min < 0: return (self - ((self.min//b)*b)) % b
+    if self.min < 0: return create_node(ModNode(create_rednode(SumNode, [self, NumNode(-1*(self.min//b)*b)]), b))
```

Comment:
I really doubt this should be calling create node

Reply:
Did you mean create "red" node? I fixed that. (Nested SumNodes resulted if `self` was a SumNode)

Removing calls to `create_node` will be ok if NumNode implements it's own `__mod__` method.